src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -1041,4 +1041,41 @@ private void repaintIfBackgroundsDiffer(JComponent comp) {
             }
         }
     }
+
+    // Toggle flag for drawing the mnemonic state
+    private static boolean isMnemonicHidden = true;
+
+    /**
+     * Sets the state of the hide mnemonic flag. This flag is used by the
+     * component UI delegates to determine if the mnemonic should be rendered.
+     * This method is a non operation if the underlying operating system
+     * does not support the mnemonic hiding feature.
+     *
+     * @param hide true if mnemonics should be hidden
+     * @since 23
+     */
```

Comment:
Hmm. So .. new public API ? Is this absolutely necessary ? 
I don't see why an app would need to call this directly. 
And it would need a CSR, and it is too late for 23 anyway.

Reply:
> Hmm. So .. new public API ? Is this absolutely necessary ?
I don't see why an app would need to call this directly.

`setMnemonicHidden` can be changed to a `protected` member as you pointed out it may not be required by an app to call this method.
But I guess the `isMnemonicHidden` should be public API.

> And it would need a CSR, and it is too late for 23 anyway.

Will update the `@since to 24` for `isMnemonicHidden` method if we are ok with `isMnemonicHidden` method's access modifier.

Reply:
Infact `isMnemonicHidden` can also be changed to a `protected` member of the class. I will check and update.

Reply:
protected members of public classes are part of the API
Go look at javadoc - or generate javadoc for this change and see for yourself. So this still requires a CSR as written.

Reply:
Yeah..I checked that and the protected members are a part of javadoc.

One possible way to not have a CSR is to remove the access modifiers from the method that results in default access specifier. In that case, these methods no longer be a part of javadoc API.